src/decorators/index.ts

```diff
@@ -18,52 +18,90 @@
  */
 
 import { Request, Response, RequestHandler, NextFunction } from 'express';
+import _ from 'lodash';
 import jwt from 'jsonwebtoken';
+import fetch from 'node-fetch';
 import { loggerFor } from '../logger';
 import { config } from '../config';
 const L = loggerFor(__filename);
 
-const getToken = (request: Request) => {
-  if (!request.headers.authorization) {
-    return undefined;
+type TokenValidationResult = { success: boolean; data?: TokenData; error?: any };
```

Comment:
i would probably restrict the types allowed as "error", no reason to let errors by any type

Reply:
I ended up removing error b/c it isn't used up the chain or returned in the request
So I just added logging wherever errors are captured and removed it from the type